uwuno/src/test/java/com/learning/uwuno/unit/roomControllerTests.java

```diff
@@ -28,10 +31,231 @@
+        ArrayList<room> roomList = new ArrayList<>();
+        room room = new room("test", false);
+        roomList.add(room);
+        when(service.getRoomList()).thenReturn(roomList);
+
+        mock.perform(get("/rooms")
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$[0]['name']").value("test"))
+                .andExpect(jsonPath("$.length()").value(1))
+                .andDo(print());
+    }
+
+    @Test
+    public void GET_room() throws Exception {
+        room room = new room("test", false);
+        when(service.getRoom(anyString())).thenReturn(room);
+        mock.perform(get("/rooms/2"))
+                .andExpect(status().isOk())
+                .andExpect(jsonPath("$.name").value(room.getName()))
+                .andExpect(jsonPath("$.uid").value(room.getUid()))
+                .andExpect(jsonPath("$.players").value(room.getPlayers()))
+                .andDo(print());
+    }
+
+    @Test
+    public void GET_room_not_found() throws Exception {
+        when(service.getRoom(anyString())).thenThrow(NoSuchElementException.class);
+        mock.perform(get("/rooms/2"))
+                .andExpect(status().isNotFound())
+                .andDo(print());
+    }
+
+    @Test
+    public void POST_add_room() throws Exception {
+        room room = new room("test", false);
+        when(service.addRoom(anyString(), anyBoolean())).thenReturn(room);
+
+        ArrayList<String> keys = new ArrayList<>();
+        keys.add("roomName");
+        keys.add("useBlankCards");
+
+        ArrayList<String> values = new ArrayList<>();
+        values.add("test");
+        values.add("false");
+
+        mock.perform(post("/rooms")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(testUtils.createJSON(keys, values))
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isOk())
+                .andExpect(content().contentType(MediaType.APPLICATION_JSON))
+                .andExpect(jsonPath("$.name").value(room.getName()))
+                .andExpect(jsonPath("$.uid").value(room.getUid()))
+                .andDo(print());
+    }
+
+    @Test
+    public void POST_add_room_internal_server_err() throws Exception {
+        String invalidJSON = "{" + "]";
+
+        mock.perform(post("/rooms")
+                .contentType(MediaType.APPLICATION_JSON)
+                .content(invalidJSON)
+                .accept(MediaType.APPLICATION_JSON))
+                .andExpect(status().isInternalServerError())
+                .andDo(print());
+    }
 
     @Test
-    public void testRoom_POST() throws Exception {
-//        this.mock.perform(post("/Rooms")).andDo(print()).andExpect(status().isOk());
+    public void POST_add_room_bad_req_err() throws Exception {
+        when(service.addRoom(anyString(), anyBoolean())).thenThrow(new com.learning.uwuno.errors.badRequest());
```

Comment:
You're not testing anything here. No matter error or no error, you're gunna get a bad request.
Recommend: Remove all thenThrow() + test for throws. The controller should throw if the wrong parameters are passed in to start with.

Reply:
It depends on the function. This test suite is designed to test the room controller, not the service. If the controller's function is just calling the service and catching the error, this test makes sense. I changed the test for the POST room function because there are actual checks within the controller function before calling service functions that would throw an error.